pkg/kafka/auth/tls_test.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+)
+
+func TestSetTLSConfiguration(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
+	saramaConfig := sarama.NewConfig()
+	tlsConfig := &tlscfg.Options{
+		Enabled: true,
+	}
+	defer tlsConfig.Close()
```

Comment:
this should be after the call to `setTLSConfiguration` and error check. At this point you just created a struct, there can be no goroutine leak yet.